Adds a Known Issues section in the README (#329)
src-d/engine: README.md

To see which languages are available, check the table of [supported languages](https://docs.sourced.tech/babelfish/languages).

## Known Issues

This section contains a list of some known issues and how they can be addressed:

- Running source{d} Engine on macOS has some performance issues:

>     SQL queries on MacOS will run slower. If your workdir contains many repositories, then the underlying gitbase component (see the [Architecture](#architecture) section) other than running queries slower, may even fail to initialize. The problem is related to the way Docker mounts volumes in macOS. See [#156](https://github.com/src-d/engine/issues/156) for more context.

Reviewer: Can gitbase _evan fail to initialize_?
when did we saw this problem? can we describe it better? maybe providing a sort of approximation of what "a big dataset" is?

Author: > Can gitbase evan fail to initialize?

Maybe @smacker can help on this as I just reported what he said [here](https://github.com/src-d/engine/issues/156#issuecomment-469325407).

> maybe providing a sort of approximation of what "a big dataset" is

That's actually a big point. Without knowing the gitbase internals I actually don't know what to say, how much it depends on the number of repos, or size of the repos or whatever. Maybe @src-d/data-retrieval could help on this.

Reviewer 2: I guess you were  looking for the help from the new awesome @src-d/data-processing team instead of @src-d/data-retrieval, gitbase is not anymore a data-retrieval's project.

Author: Thank you @mcarmonaa.

>     SQL queries on MacOS will run slower. If your workdir contains many repositories, then the underlying Gitbase component (see the [Architecture](#architecture) section) other then running queries slower, may even fail to initialize. The problem seems to be related to FS mapping into Docker for MacOS. Some context can be found here [#156](https://github.com/src-d/engine/issues/156).

Reviewer 3: ```suggestion
    SQL queries on MacOS will run slower. If your workdir contains many repositories, then the underlying gitbase component (see the [Architecture](#architecture) section) other than running queries slower, may even fail to initialize. The problem is related to the way Docker mounts volumes in macOS. See [#156](https://github.com/src-d/engine/issues/156) for more context.
```
